_re2.cc

```diff
@@ -365,40 +380,6 @@ create_regexp(PyObject* self, PyObject* pattern, PyObject* error_class)
     return NULL;
   }
 
-  PyObject* groupindex = PyDict_New();
-  if (groupindex == NULL) {
-    Py_DECREF(regexp);
-    return NULL;
-  }
-
-  // Build up the attr_dict early so regexp can take ownership of our reference
-  // to groupindex.
-  regexp->attr_dict = Py_BuildValue("{sisNsO}",
-      "groups", regexp->re2_obj->NumberOfCapturingGroups(),
-      "groupindex", groupindex,
```

Comment:
Does this mean that groupindex no longer works?  In case it wasn't clear before, I don't think this change should be breaking any functionality that currently works.

Reply:
I will address that. 

I will do some checks to see if the attr_dict is performant enough otherwise I'll just extend the `RegexObject2` definition similiar to `MatrchObject2`